src/HTTPServer/HTTPMessage.cpp

```diff
@@ -210,7 +214,7 @@ size_t cHTTPRequest::ParseRequestLine(const char * a_Data, size_t a_Size)
 				}
 				m_Method = m_IncomingHeaderData.substr(LineStart, MethodEnd - LineStart);
 				m_URL = m_IncomingHeaderData.substr(MethodEnd + 1, URLEnd - MethodEnd - 1);
-				return i + 1;
+				return i + 1 - inBufferSoFar;
```

Comment:
This is the main point of the change - `i` counts over the buffer, but we want to return the number of bytes consumed from the current call only, hence the subtraction of the old buffer size.